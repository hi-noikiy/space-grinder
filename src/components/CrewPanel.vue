<template>

  <div class="columns">
    <div class="container crew-select" v-if="showSelectionPanel"  v-on:click="hideCrewSelectionPanel"></div>
        <figure class="avatar avatar-xl mx-1 crew-avatar" v-for="member in crew" v-bind:key="member.id" v-on:click="showCrewSelectionPanel">
           <img v-bind:src="member.image" alt="...">
         </figure>
        </div>
</template>

<script>
import { mapGetters } from 'vuex'
export default {
  name: 'CrewPanel',
  data () {
    return {
      showSelectionPanel: false
    }
  },
  methods: {
    showCrewSelectionPanel: function () {
      this.showSelectionPanel = true
    },
    hideCrewSelectionPanel: function () {
      this.showSelectionPanel = false
    }
  },
  computed: mapGetters(['crew'])
}
</script>

<style scoped>

</style>
