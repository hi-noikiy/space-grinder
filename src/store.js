import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

const state = {
  distance: 0,
  cellVoltage: 2,
  cvMultiplier: 0,
  ampHourCapacity: 2,
  ahcMultiplier: 0,
  coulombs: 0,
  mass: 200000,
  minerals: 100000,
  crew: [
    {id: 0, image: './assets/images/head.png'},
    {id: 1, image: '../assets/images/head2.png'},
    {id: 2, image: '../assets/images/head.png'},
    {id: 3, image: '../assets/images/head.png'},
    {id: 4, image: '../assets/images/head.png'}
  ],
  generatorUpgrades: [
    {id: 1, name: 'gen upg 1', baseCost: 10, upgCount: 0, baseProd: 2},
    {id: 2, name: 'gen upg 2', baseCost: 125, upgCount: 0, baseProd: 6},
    {id: 3, name: 'gen upg 3', baseCost: 600, upgCount: 0, baseProd: 20},
    {id: 4, name: 'gen upg 4', baseCost: 1800, upgCount: 0, baseProd: 65}
  ],
  storageUpgrades: [
    {id: 1, name: 'stor upg 1', baseCost: 10 * 10, upgCount: 0, voltIncrease: 2, ampIncrease: 2},
    {id: 2, name: 'stor upg 2', baseCost: 125 * 10, upgCount: 0, voltIncrease: 6, ampIncrease: 6},
    {id: 3, name: 'stor upg 3', baseCost: 600 * 10, upgCount: 0, voltIncrease: 20, ampIncrease: 20},
    {id: 4, name: 'stor upg 4', baseCost: 1800 * 10, upgCount: 0, voltIncrease: 65, ampIncrease: 65}
  ],
  hullUpgrades: [
    {id: 1, name: 'hull upg 1', baseCost: 10 * 100, upgCount: 0, massReduction: 20},
    {id: 2, name: 'hull upg 2', baseCost: 125 * 100, upgCount: 0, massReduction: 60},
    {id: 3, name: 'hull upg 3', baseCost: 600 * 100, upgCount: 0, massReduction: 200},
    {id: 4, name: 'hull upg 4', baseCost: 1800 * 100, upgCount: 0, massReduction: 650}
  ]
}

var addCoulombs = function (state, amount) {
  if (state.coulombs + amount < state.ampHourCapacity * 3600) {
    state.coulombs += amount
  } else { state.columbs = state.ampHourCapacity * 3600 }

  if (state.coulombs > 500) {
    state.minerals += (amount / 4)
  }
}

const mutations = {
  incrementCoulombs (state, {amount}) {
    state.generatorUpgrades.map(u => { amount += u.baseProd * u.upgCount })
    addCoulombs(state, amount)
  },
  incrementCoulombsClick (state, {amount}) {
    addCoulombs(state, amount)
  },
  incrementVolts (state, {amount}) {
    state.cellVoltage += amount
  },
  incrementAmps (state, {amount}) {
    state.ampHourCapacity += amount
  },
  incrementMass (state, {amount}) {
    state.mass += amount
  },
  decrementMass (state, {amount}) {
    state.mass -= amount
  },
  incrementMinerals (state, {amount}) {
    state.minerals += amount
  },
  decrementMinerals (state, {amount}) {
    state.minerals -= amount
  },
  buyGeneratorUpgrades (state, {id}) {
    console.log('buyGeneratorUpgrades')
    state.generatorUpgrades.map((upg) => {
      if (upg.id === id && upg.baseCost * Math.pow(1.15, upg.upgCount) < state.minerals) {
        state.minerals -= upg.baseCost * Math.pow(1.15, upg.upgCount)
        upg.upgCount++
      }
    })
  },
  buyStorageUpgrades (state, {id}) {
    console.log('buyStorageUpgrades')
    state.storageUpgrades.map((upg) => {
      if (upg.id === id && upg.baseCost * Math.pow(1.15, upg.upgCount) < state.minerals) {
        state.minerals -= upg.baseCost * Math.pow(1.15, upg.upgCount)
        upg.upgCount++
      }
    })
  },
  buyHullUpgrades (state, {id}) {
    console.log('buyHullUpgrades')
    state.hullUpgrades.map((upg) => {
      if (upg.id === id && upg.baseCost * Math.pow(1.15, upg.upgCount) < state.minerals) {
        state.minerals -= upg.baseCost * Math.pow(1.15, upg.upgCount)
        upg.upgCount++
      }
    })
  }
}

const actions = {
  incrementCoulombs: ({ commit }, amount) => commit('incrementCoulombs', {amount: amount}),
  incrementCoulombsClick: ({ commit }, amount) => commit('incrementCoulombsClick', {amount: amount}),
  incrementVolts: ({ commit }, amount) => commit('incrementVolts', {amount: amount}),
  incrementMass: ({ commit }, amount) => commit('incrementMass', {amount: amount}),
  decrementMass: ({ commit }, amount) => commit('decrementMass', {amount: amount}),
  buyGeneratorUpgrades: ({ commit }, id) => commit('buyGeneratorUpgrades', {id: id}),
  buyStorageUpgrades: ({ commit }, id) => commit('buyStorageUpgrades', {id: id}),
  buyHullUpgrades: ({ commit }, id) => commit('buyHullUpgrades', {id: id})
}

const getters = {
  generatorUpgrades: state => state.generatorUpgrades,
  storageUpgrades: state => state.storageUpgrades,
  hullUpgrades: state => state.hullUpgrades,
  minerals: state => state.minerals,
  capacity: state => state.ampHourCapacity * 3600,
  joules: state => state.cellVoltage * state.coulombs,
  distance: state => Math.tanh((state.cellVoltage * state.coulombs) / state.mass),
  voltage: state => {
    state.storageUpgrades.map(u => { state.cellVoltage += u.voltIncrease * u.upgCount })
    return state.cellVoltage
  },
  amps: state => {
    state.storageUpgrades.map(u => { state.ampHourCapacity += u.ampIncrease * u.upgCount })
    return state.ampHourCapacity
  },
  mass: state => {
    state.hullUpgrades.map(u => { state.mass -= u.massReduction * u.upgCount })
    return state.mass
  },
  crew: state => state.crew
}

export default new Vuex.Store({
  state,
  getters,
  actions,
  mutations
})
