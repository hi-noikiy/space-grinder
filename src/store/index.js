import Vue from 'vue'
import Vuex from 'vuex'
import crew from './modules/crew'
import ship from './modules/ship'
import upgrades from './modules/upgrades'
import liquidgas from './modules/liquidgas'
import research from './modules/research'

Vue.use(Vuex)

export default new Vuex.Store({
  modules: {
    crew,
    ship,
    upgrades,
    liquidgas,
    research
  }
})
