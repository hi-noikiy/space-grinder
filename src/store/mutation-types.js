export const ADD_TO_CART = 'ADD_TO_CART'
export const INCREMENT_COULOMBS = 'INCREMENT_COULOMBS'
export const INCREMENT_AMPS = 'INCREMENT_AMPS'
export const INCREMENT_MINERALS = 'INCREMENT_MINERALS'
export const DECREMENT_MINERALS = 'DECREMENT_MINERALS'
export const INCREMENT_COULOMBS_CLICK = 'INCREMENT_COULOMBS_CLICK'
export const INCREMENT_VOLTS = 'INCREMENT_VOLTS'
export const INCREMENT_MASS = 'INCREMENT_MASS'
export const DECREMENT_MASS = 'DECREMENT_MASS'
export const BUY_GENERATOR_UPGRADES = 'BUY_GENERATOR_UPGRADES'
export const BUY_UPGRADES = 'BUY_UPGRADES'
export const BUY_STORAGE_UPGRADES = 'BUY_STORAGE_UPGRADES'
export const BUY_HULL_UPGRADES = 'BUY_HULL_UPGRADES'
export const ASSIGN_CREWMEMBER = 'ASSIGN_CREWMEMBER'
export const BUY_UPGRADE_UPGRADE = 'BUY_UPGRADE_UPGRADE'
export const SET_VOLTS = 'SET_VOLTS'
export const SET_AMPS = 'SET_AMPS'
export const DECREMENT_HYDROGEN = 'DECREMENT_HYDROGEN'
export const DECREMENT_OXYGEN = 'DECREMENT_OXYGEN'
export const DECREMENT_CARBON = 'DECREMENT_CARBON'
export const INCREMENT_HYDROGEN = 'INCREMENT_HYDROGEN'
export const INCREMENT_OXYGEN = 'INCREMENT_OXYGEN'
export const INCREMENT_CARBON = 'INCREMENT_CARBON'
