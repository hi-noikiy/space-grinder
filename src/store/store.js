import Vue from 'vue'
import Vuex from 'vuex'
import * as types from './mutation-types'
import VuexPersist from 'vuex-persist'

Vue.use(Vuex)

const vuexLocalStorage = new VuexPersist({
  key: 'vuex', // The key to store the state on in the storage provider.
  storage: window.localStorage // or window.sessionStorage or localForage
  // Function that passes the state and returns the state with only the objects you want to store.
  // reducer: state => state,
  // Function that passes a mutation and lets you decide if it should update the state in localStorage.
  // filter: mutation => (true)
})

const state = {
  distance: 0,
  cellVoltage: 2,
  cvMultiplier: 0,
  ampHourCapacity: 2,
  ahcMultiplier: 0,
  coulombs: 0,
  mass: 200000,
  minerals: 100,
  crew: [
    {id: 10, image: 'def', desc: 'select a crewmember', name: 'Select member'},
    {id: 11, image: 'def', desc: 'select a crewmember, unlocks at ?', name: 'Select member'},
    {id: 12, image: 'def', desc: 'select a crewmember, unlocks at ?', name: 'Select member'},
    {id: 13, image: 'def', desc: 'select a crewmember, unlocks at ?', name: 'Select member'},
    {id: 14, image: 'def', desc: 'select a crewmember, unlocks at ?', name: 'Select member'}
  ],
  availableCrew: [
    {id: 20, image: 'av1', selected: false, desc: 'this is a description of the selected crewmember', name: 'Crewmember 1', prof: 'Mechanic'},
    {id: 21, image: 'av2', selected: false, desc: 'this is a description of the selected crewmember', name: 'Crewmember 1', prof: 'Soldier'},
    {id: 22, image: 'av3', selected: false, desc: 'this is a description of the selected crewmember', name: 'Crewmember 1', prof: 'Botanist'},
    {id: 23, image: 'av4', selected: false, desc: 'this is a description of the selected crewmember', name: 'Crewmember 1', prof: 'Physicist '},
    {id: 24, image: 'av5', selected: false, desc: 'this is a description of the selected crewmember', name: 'Crewmember 1', prof: 'Deadbeat'}
  ],
  generatorUpgrades: [
    {id: 1, name: 'gen upg 1', baseCost: 10, upgCount: 0, baseProd: 2, level: 1},
    {id: 2, name: 'gen upg 2', baseCost: 125, upgCount: 0, baseProd: 6, level: 1},
    {id: 3, name: 'gen upg 3', baseCost: 600, upgCount: 0, baseProd: 20, level: 1},
    {id: 4, name: 'gen upg 4', baseCost: 1800, upgCount: 0, baseProd: 65, level: 1}
  ],
  storageUpgrades: [
    {id: 1, name: 'stor upg 1', baseCost: 10 * 10, upgCount: 0, voltIncrease: 2, ampIncrease: 2, level: 1},
    {id: 2, name: 'stor upg 2', baseCost: 125 * 10, upgCount: 0, voltIncrease: 6, ampIncrease: 6, level: 1},
    {id: 3, name: 'stor upg 3', baseCost: 600 * 10, upgCount: 0, voltIncrease: 20, ampIncrease: 20, level: 1},
    {id: 4, name: 'stor upg 4', baseCost: 1800 * 10, upgCount: 0, voltIncrease: 65, ampIncrease: 65, level: 1}
  ],
  hullUpgrades: [
    {id: 1, name: 'hull upg 1', baseCost: 10 * 100, upgCount: 0, massReduction: 20, level: 1},
    {id: 2, name: 'hull upg 2', baseCost: 125 * 100, upgCount: 0, massReduction: 60, level: 1},
    {id: 3, name: 'hull upg 3', baseCost: 600 * 100, upgCount: 0, massReduction: 200, level: 1},
    {id: 4, name: 'hull upg 4', baseCost: 1800 * 100, upgCount: 0, massReduction: 650, level: 1}
  ]
}

var addCoulombs = function (state, amount) {
  if (state.coulombs + amount < state.ampHourCapacity * 3600) {
    state.coulombs += amount
  } else { state.columbs = state.ampHourCapacity * 3600 }

  if (state.coulombs > 500) {
    state.minerals += (amount / 4)
  }
}

const mutations = {
  [types.INCREMENT_COULOMBS] (state, {amount}) {
    state.generatorUpgrades.map(u => { amount += (u.baseProd * u.upgCount * u.level) })
    addCoulombs(state, amount)
  },
  [types.INCREMENT_COULOMBS_CLICK] (state, {amount}) {
    addCoulombs(state, amount)
  },
  [types.INCREMENT_VOLTS] (state, {amount}) {
    state.cellVoltage += amount
  },
  [types.INCREMENT_AMPS] (state, {amount}) {
    state.ampHourCapacity += amount
  },
  [types.INCREMENT_MASS] (state, {amount}) {
    state.mass += amount
  },
  [types.DECREMENT_MASS] (state, {amount}) {
    state.mass -= amount
  },
  [types.INCREMENT_MINERALS] (state, {amount}) {
    state.minerals += amount
  },
  [types.DECREMENT_MINERALS] (state, {amount}) {
    state.minerals -= amount
  },
  [types.BUY_GENERATOR_UPGRADES] (state, {id}) {
    // console.log('buyGeneratorUpgrades')
    state.generatorUpgrades.map((upg) => {
      if (upg.id === id && upg.baseCost * Math.pow(1.15, upg.upgCount) < state.minerals) {
        state.minerals -= upg.baseCost * Math.pow(1.15, upg.upgCount)
        upg.upgCount++
      }
    })
  },
  [types.BUY_STORAGE_UPGRADES] (state, {id}) {
    // console.log('buyStorageUpgrades')
    state.storageUpgrades.map((upg) => {
      if (upg.id === id && upg.baseCost * Math.pow(1.15, upg.upgCount) < state.minerals) {
        state.minerals -= upg.baseCost * Math.pow(1.15, upg.upgCount)
        upg.upgCount++
      }
    })
  },
  [types.BUY_HULL_UPGRADES] (state, {id}) {
    // console.log('buyHullUpgrades')
    state.hullUpgrades.map((upg) => {
      if (upg.id === id && upg.baseCost * Math.pow(1.15, upg.upgCount) < state.minerals) {
        state.minerals -= upg.baseCost * Math.pow(1.15, upg.upgCount)
        upg.upgCount++
      }
    })
  },
  [types.ASSIGN_CREWMEMBER] (state, {slotIdCrewId}) {
    console.log(slotIdCrewId['slotId'])
    console.log(slotIdCrewId['crewId'])
    if (!state.availableCrew[slotIdCrewId['crewId']].selected) {
      state.crew[slotIdCrewId['slotId']] = state.availableCrew[slotIdCrewId['crewId']]
      state.availableCrew[slotIdCrewId['crewId']].selected = true
    }
  }
}

const actions = {
  incrementCoulombs: ({ commit }, amount) => commit(types.INCREMENT_COULOMBS, {amount: amount}),
  incrementCoulombsClick: ({ commit }, amount) => commit(types.INCREMENT_COULOMBS_CLICK, {amount: amount}),
  incrementVolts: ({ commit }, amount) => commit(types.INCREMENT_VOLTS, {amount: amount}),
  incrementMass: ({ commit }, amount) => commit(types.INCREMENT_MASS, {amount: amount}),
  decrementMass: ({ commit }, amount) => commit(types.DECREMENT_MASS, {amount: amount}),
  buyGeneratorUpgrades: ({ commit }, id) => commit(types.BUY_GENERATOR_UPGRADES, {id: id}),
  buyStorageUpgrades: ({ commit }, id) => commit(types.BUY_STORAGE_UPGRADES, {id: id}),
  buyHullUpgrades: ({ commit }, id) => commit(types.BUY_HULL_UPGRADES, {id: id}),
  assignCrewMember: ({ commit }, slotIdCrewId) => commit(types.ASSIGN_CREWMEMBER, {slotIdCrewId: slotIdCrewId})
}

const getters = {
  generatorUpgrades: state => state.generatorUpgrades,
  storageUpgrades: state => state.storageUpgrades,
  hullUpgrades: state => state.hullUpgrades,
  minerals: state => state.minerals,
  capacity: state => state.ampHourCapacity * 3600,
  joules: state => state.cellVoltage * state.coulombs,
  distance: state => Math.tanh((state.cellVoltage * state.coulombs) / state.mass),
  voltage: state => {
    state.storageUpgrades.map(u => { state.cellVoltage += u.voltIncrease * u.upgCount })
    return state.cellVoltage
  },
  amps: state => {
    state.storageUpgrades.map(u => { state.ampHourCapacity += u.ampIncrease * u.upgCount })
    return state.ampHourCapacity
  },
  mass: state => {
    state.hullUpgrades.map(u => { state.mass -= u.massReduction * u.upgCount })
    return state.mass
  },
  crew: state => state.crew,
  availableCrew: state => state.availableCrew
}

export default new Vuex.Store({
  plugins: [vuexLocalStorage.plugin],
  state,
  getters,
  actions,
  mutations
})
