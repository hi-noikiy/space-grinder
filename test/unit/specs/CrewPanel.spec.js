import { shallow } from '@vue/test-utils'
import CrewPanel from '@/components/CrewPanel'
import store from '@/store'
import { createRenderer } from 'vue-server-renderer'

describe('CrewPanel.vue', () => {
  it('matches snapshot', () => {
    const renderer = createRenderer()
    const wrapper = shallow(CrewPanel, {store})

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
