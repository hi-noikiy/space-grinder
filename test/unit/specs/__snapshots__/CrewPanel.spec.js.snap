// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CrewPanel.vue matches snapshot 1`] = `
<div class="columns crew-panel">
    <div class="column col-4">
        <!---->
    </div>
    <div class="column col-8">
        <div class="accordion-body">
            <div class=" popover popover-left">
                <div class="tile tile-centered canBuy">
                    <div class="tile-icon">
                        <figure class="avatar avatar-lg mx-1">
                            <img src="default.png" alt="select a crewmember">
                        </figure>
                    </div>
                    <div class="tile-content">
                        <div class="tile-title text-bold">Select member</div>
                        <!---->
                        <!---->
                        <div class="tile-subtitle ">Unlocked</div>
                    </div>
                    <div class="tile-action">
                        <button class="btn btn-link">
                            <i class="icon icon-plus"></i>
                        </button>
                    </div>
                </div>
                <!---->
            </div>
            <div class=" popover popover-left">
                <div class="tile tile-centered">
                    <div class="tile-icon">
                        <figure class="avatar avatar-lg mx-1">
                            <img src="default.png" alt="select a crewmember">
                        </figure>
                    </div>
                    <div class="tile-content">
                        <div class="tile-title text-bold">Select member</div>
                        <!---->
                        <div class="tile-subtitle ">0.10416666666666667 %</div>
                        <!---->
                    </div>
                    <div class="tile-action">
                        <button class="btn btn-link disabled">
                            <i class="icon icon-plus"></i>
                        </button>
                    </div>
                </div>
                <!---->
            </div>
            <div class=" popover popover-left">
                <div class="tile tile-centered">
                    <div class="tile-icon">
                        <figure class="avatar avatar-lg mx-1">
                            <img src="default.png" alt="select a crewmember">
                        </figure>
                    </div>
                    <div class="tile-content">
                        <div class="tile-title text-bold">Select member</div>
                        <!---->
                        <div class="tile-subtitle ">0.0010416666666666667 %</div>
                        <!---->
                    </div>
                    <div class="tile-action">
                        <button class="btn btn-link disabled">
                            <i class="icon icon-plus"></i>
                        </button>
                    </div>
                </div>
                <!---->
            </div>
            <div class=" popover popover-left">
                <div class="tile tile-centered">
                    <div class="tile-icon">
                        <figure class="avatar avatar-lg mx-1">
                            <img src="default.png" alt="select a crewmember">
                        </figure>
                    </div>
                    <div class="tile-content">
                        <div class="tile-title text-bold">Select member</div>
                        <!---->
                        <div class="tile-subtitle ">0.000010416666666666668 %</div>
                        <!---->
                    </div>
                    <div class="tile-action">
                        <button class="btn btn-link disabled">
                            <i class="icon icon-plus"></i>
                        </button>
                    </div>
                </div>
                <!---->
            </div>
            <div class=" popover popover-left">
                <div class="tile tile-centered">
                    <div class="tile-icon">
                        <figure class="avatar avatar-lg mx-1">
                            <img src="default.png" alt="select a crewmember">
                        </figure>
                    </div>
                    <div class="tile-content">
                        <div class="tile-title text-bold">Select member</div>
                        <!---->
                        <div class="tile-subtitle ">1.0416666666666667e-7 %</div>
                        <!---->
                    </div>
                    <div class="tile-action">
                        <button class="btn btn-link disabled">
                            <i class="icon icon-plus"></i>
                        </button>
                    </div>
                </div>
                <!---->
            </div>
        </div>
    </div>
</div>
`;
